core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -338,6 +340,9 @@ private[spark] class Executor(
             metricsSystem = env.metricsSystem)
           threwException = false
           res
+        } catch {
+          case _: CommitDeniedException =>
+            throw new TaskKilledException("commit denied")
```

Comment:
Maybe we should throw a more specific exception for the `already committed` case ? Executor can know about the `already committed` case before it sends the statusUpdate, so 
we do not need to wait until the driver's statusUpdate.

It also seems ok to fix it up on the driver, I'll try it tomorrow.